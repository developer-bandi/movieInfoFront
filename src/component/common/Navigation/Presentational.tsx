import { useState } from 'react';
import { useLocation } from 'react-router-dom';
import styles from './Style';

const Navigation = () => {
  const location = useLocation();
  const [templocation, setTempLocation] = useState(location.pathname);
  return (
    <styles.MainBlock active={location.pathname === '/' ? 'false' : 'true'}>
      <styles.MenuListBlock>
        {menulist.map((data, index) => {
          return (
            <styles.MenuBlock key={menulinklist[index]}>
              <styles.Menu
                active={templocation === menulinklist[index] ? 'true' : 'false'}
                to={menulinklist[index]}
                onMouseOver={() => {
                  setTempLocation(menulinklist[index]);
                }}
                onMouseOut={() => {
                  setTempLocation(location.pathname);
                }}
              >
                {data}
              </styles.Menu>
            </styles.MenuBlock>
          );
        })}
      </styles.MenuListBlock>
    </styles.MainBlock>
  );
};

const menulist: string[] = ['홈', '랭킹', '영화검색', '나만의 영화'];
const menulinklist: string[] = ['/', '/sortedmovie', '/search', '/favorites'];

export default Navigation;
