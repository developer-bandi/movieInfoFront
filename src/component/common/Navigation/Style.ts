import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { css } from 'styled-components';

const MainBlock = styled.nav<{ active: string }>`
  width: 100%;
  height: 4rem;
  @media screen and (max-width: 500px) {
    height: 3rem;
  }

  ${(props) =>
    props.active === 'true' &&
    css`
      border-bottom: 1px solid #e8e8e8;
    `}
`;

const MenuListBlock = styled.ul`
  display: flex;
  justify-content: center;
  list-style: none;
  padding-left: 0;
  margin: 0;
`;

const MenuBlock = styled.li`
  background-color: transparent;
  font-weight: bold;
  font-size: 25px;
  margin-top: 1rem;
  flex-basis: 25%;
  text-align: center;

  @media screen and (max-width: 500px) {
    font-size: 15px;
  }
`;

const Menu = styled(Link)<{ active: string }>`
  display: block;
  text-decoration: none;
  width: 100%;
  height: 3rem;
  color: gray;
  &:hover {
    color: black;
    border-bottom: 3px solid black;
  }

  ${(props) =>
    props.active === 'true' &&
    css`
      color: black;
      border-bottom: 3px solid black;
    `}

  @media screen and (max-width: 500px) {
    height: 2rem;
  }
`;

const styles = { MainBlock, MenuListBlock, MenuBlock, Menu };

export default styles;
