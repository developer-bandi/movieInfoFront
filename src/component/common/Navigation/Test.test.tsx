import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from './Presentational';

describe('<Navigation/> Presentational', () => {
  it('전체 화면 검증', () => {
    const utils = render(<Navigation />, { wrapper: MemoryRouter });
    expect(utils.container).toMatchSnapshot();
  });
});
