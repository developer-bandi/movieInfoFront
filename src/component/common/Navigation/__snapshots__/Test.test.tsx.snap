// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> Presentational Navigation 스냅샷 비교 1`] = `
<div>
  <nav
    class="sc-bdvvtL kPLDXe"
  >
    <ul
      class="sc-gsDKAQ cpbHdK"
    >
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="true"
          class="sc-hKwDye iDZeQi"
          href="/"
        >
          홈
        </a>
      </li>
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="false"
          class="sc-hKwDye eYjUNd"
          href="/sortedmovie"
        >
          랭킹
        </a>
      </li>
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="false"
          class="sc-hKwDye eYjUNd"
          href="/search"
        >
          영화검색
        </a>
      </li>
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="false"
          class="sc-hKwDye eYjUNd"
          href="/favorites"
        >
          나만의 영화
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<Navigation/> Presentational 전체 화면 검증 1`] = `
<div>
  <nav
    class="sc-bdvvtL kPLDXe"
  >
    <ul
      class="sc-gsDKAQ cpbHdK"
    >
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="true"
          class="sc-hKwDye iDZeQi"
          href="/"
        >
          홈
        </a>
      </li>
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="false"
          class="sc-hKwDye eYjUNd"
          href="/sortedmovie"
        >
          랭킹
        </a>
      </li>
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="false"
          class="sc-hKwDye eYjUNd"
          href="/search"
        >
          영화검색
        </a>
      </li>
      <li
        class="sc-dkPtRN iBhiyQ"
      >
        <a
          active="false"
          class="sc-hKwDye eYjUNd"
          href="/favorites"
        >
          나만의 영화
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
